packages/amplify_datastore/ios/Classes/types/model/FlutterSerializedModel.swift

```diff
@@ -250,19 +319,66 @@ struct FlutterSerializedModel: Model, JSONValueHolder {
                     case .some(.number(let deserializedValue)) = values[key] {
                 
                 result[key] = NSNumber(value: Int(deserializedValue) )
-            } else {
+            }
+            else if case .embedded(_, .some(let customTypeScheam)) = field?.type,
+                case .some(.object(let deserializedValue)) = values[key] {
+                let customTypeName = customTypeScheam.name
+                result[key] = [
+                    "customTypeName": customTypeName,
+                    "serializedData": try FlutterSerializedModel.generateSerializedJsonData(
+                        values: deserializedValue,
+                        modelSchemaRegistry: modelSchemaRegistry,
+                        customTypeSchemaRegistry: customTypeSchemaRegistry,
+                        modelName: customTypeName
+                    )
+                ]
+            } else if case .embeddedCollection = field?.type {
```

Comment:
Nit: This whole thing kind of seems like it could be a switch on `field?.type`?

Reply:
Ditto.